liquibase-core/src/main/java/liquibase/changelog/ChangeLogParameters.java

```diff
@@ -117,21 +118,25 @@ public void set(String key, String value, String contexts, String labels, String
     public void set(String key, String value, ContextExpression contexts, Labels labels, String databases,
                     boolean globalParam, DatabaseChangeLog changeLog) {
         /**
-         * TODO: this was a bug. Muliple created parameters have been created, but the corresponding method in
+         * TODO: this was a bug. Multiple created parameters have been created, but the corresponding method in
          * #findParameter() is only catching the first one. So here we should eliminate duplicate entries
          **/
         if (globalParam) {
-            // if it is global param ignore additional adds
+            // if it is global param remove duplicate non-global parameters
             ChangeLogParameter param = findParameter(key, null);
-            if (param == null) {
-                // okay add it
-                changeLogParameters.add(new ChangeLogParameter(key, value, contexts, labels, databases, globalParam,
-                    changeLog));
+            if (param != null && ! param.isGlobal()) {
+                changeLogParameters.remove(param);
             }
-        } else {
-            //this is a non-global param, just add it
+            // okay add it
             changeLogParameters.add(new ChangeLogParameter(key, value, contexts, labels, databases, globalParam,
                 changeLog));
+        } else {
+           ChangeLogParameter param = findParameter(key, changeLog);
```

Comment:
Duplication of code, can we move the following lines to a method and call in all three places?
           ChangeLogParameter param = findParameter(key, changeLog);

if (param != null && ! param.isGlobal() && param.getChangeLog() == changeLog) {
               changeLogParameters.remove(param);
           }